packages/web3-react/src/hooks/useDisconnect.ts

```diff
@@ -12,16 +13,16 @@ export const useForceDisconnect = (): {
 } => {
   const { deactivate, connector } = useWeb3();
   const extendedConnector = connector as ExtendedConnector;
+  const { disconnectAsync: wagmiDisconnect } = useDisconnectWagmi();
 
   const disconnect = useCallback(async () => {
-    try {
-      deactivate();
-      extendedConnector?.deactivate();
-      await extendedConnector?.close?.();
-    } catch (error) {
-      //
-    }
-  }, [deactivate, extendedConnector]);
+    // disconnect wallets connected through web3-react connectors
+    deactivate();
+    extendedConnector?.deactivate();
+    await extendedConnector?.close?.();
```

Comment:
can this call throw? so `wagmiDisconnect` will be prevented.
If yes, better to use await `Promise.all`

Reply:
No, the `close` method just emits an event. Actually, it should not be a Promise, but it is defined as async method in the original web3-react library (without awaiting anything in the method's body).
This code will hopefully become obsolete soon, so I'm going to leave it as is.